enhancements/console/samples.md

```diff
@@ -0,0 +1,309 @@
+
+The user can select one of the many samples:
+
+![Project access tab with role select field to select an access role](samples.png)
+
+Currently, this opens a form to import a git repository. Other sample types aren't supported at the moment.
+
+## Motivation
+
+Creating a `ConsoleSample` CRD will promote the samples feature in the Developer Console to an extendable feature similar to QuickStarts. This makes it much easier for other teams to provide their own samples, for example with their operator.
+
+Samples are no longer tied to the exact OpenShift version and could be updated with the operator. Also customers can add their own samples if needed.
+
+This will also allow us to migrate from the (deprecated) [samples-operator](https://github.com/openshift/cluster-samples-operator) to a simple custom resource in the future and to add other sample types in the future.
+
+### User Stories
+
+* As an OpenShift Serverless engineer, I want to provide additional samples that are tied to an operator version, not an OpenShift release.
+* As a cluster administrator, I want to add additional samples to the Developer Console sample catalog.
+
+### Goals
+
+1. This feature should allow other teams to provide their Developer Console samples without applying them to the sample-operator or the console-operator.
+2. Samples should be allowed to localize/translated.
+
+### Non-Goals
+
+1. We don't want to turn off the support for the (deprecated) [samples-operator](https://github.com/openshift/cluster-samples-operator) right now. So the console will not provide any samples on its own (in 4.14).
+2. A way to hide installed samples. Operators creating a sample resource should have a way to opt out of their samples.
+
+## Proposal
+
+### Workflow Description
+
+The Console Operator will provide a cluster-wide `ConsoleSample` CRD.
+
+Other operators like the OpenShift Serverless operator will install their own `ConsoleSample` CRs.
+
+The web console will automatically load all `ConsoleSample` CRs and show them in the sample catalog.
+
+Users can select one of samples and follow the import flows to create a demo application.
+
+### API Extensions
+
+The console operator automatically installs a new `ConsoleSample` CRD.
+
+### Risks and Mitigations
+
+#### The OpenShift Console is optional and could be disabled
+
+Other operators should expect that the `ConsoleSample` CRD is unavailable like the other `Console*` CRDs.
+
+### Drawbacks
+
+Another extra (but specific) custom resource just for console samples.
+
+## Design Details
+
+An "Import from Git" example of how we have it today as part of the Builder Images annotation:
+
+```yaml
+apiVersion: console.openshift.io/v1
+kind: ConsoleSample
+metadata:
+  name: java-sample
+spec:
+  title: Java
+  description: |
+    Build and run Java applications using Maven and OpenJDK.
+    Sample repository: https://github.com/jboss-openshift/openshift-quickstarts
+  icon: base64 encoded image
+  provider: Red Hat
```

Comment:
Do we worry about small differences in the provider name? (`Red Hat`, `Red Hat, Inc.`, `Red Hat, Inc`, `redhat`, etc?)

Hard to get around, I guess. We might simply have to document recommendations somewhere so teams are consistent.

Reply:
As written above, I added this explanation below the sample and will incl. this similar in the API ~= CRD.

> 4. `provider` like "Red Hat" is optional and can have a maximum length of **20** characters.
> 
>    Currently, the provider is only shown on the sample card tile below the title with the prefix "Provided by " (see screenshot above).